src/rdbms-connect/azext_rdbms_connect/_client_factory.py

```diff
@@ -0,0 +1,159 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
+
+RM_URI_OVERRIDE = 'AZURE_CLI_RDBMS_RM_URI'
+SUB_ID_OVERRIDE = 'AZURE_CLI_RDBMS_SUB_ID'
+CLIENT_ID = 'AZURE_CLIENT_ID'
+TENANT_ID = 'AZURE_TENANT_ID'
+CLIENT_SECRET = 'AZURE_CLIENT_SECRET'
+
+
+def get_mariadb_management_client(cli_ctx, **_):
+    from os import getenv
+    from azure.mgmt.rdbms.mariadb import MariaDBManagementClient
+
+    # Allow overriding resource manager URI using environment variable
+    # for testing purposes. Subscription id is also determined by environment
+    # variable.
+    rm_uri_override = getenv(RM_URI_OVERRIDE)
+    if rm_uri_override:
+        client_id = getenv(CLIENT_ID)
+        if client_id:
+            from azure.common.credentials import ServicePrincipalCredentials
+            credentials = ServicePrincipalCredentials(
+                client_id=client_id,
+                secret=getenv(CLIENT_SECRET),
+                tenant=getenv(TENANT_ID))
+        else:
+            from msrest.authentication import Authentication  # pylint: disable=import-error
+            credentials = Authentication()
+
+        return MariaDBManagementClient(
+            subscription_id=getenv(SUB_ID_OVERRIDE),
+            base_url=rm_uri_override,
+            credentials=credentials)
```

Comment:
recommend to use CLI core frame work for authenticating the client. see
https://github.com/Azure/azure-cli/blob/027c92d91e08d3a63815d50f76b8316d2ad41efc/src/azure-cli-core/azure/cli/core/commands/client_factory.py#L50-L75